src/snowflake/snowpark/modin/plugin/_internal/binary_op_utils.py

```diff
@@ -385,32 +429,102 @@ def compute_binary_op_between_snowpark_columns(
         )
     elif _op_is_between_two_timedeltas_or_timedelta_and_null(
         first_datatype(), second_datatype()
-    ) and op in ("eq", "ne", "gt", "ge", "lt", "le", "truediv"):
+    ) and op in (
+        "eq",
+        "ne",
+        "gt",
+        "ge",
+        "lt",
+        "le",
+    ):
         # These operations, when done between timedeltas, work without any
         # extra handling in `snowpark_pandas_type` or `binary_op_result_column`.
-        # They produce outputs that are not timedeltas (e.g. numbers for floordiv
-        # and truediv, and bools for the comparisons).
         pass
+    elif op == "mul" and (
```

Comment:
the readability of `compute_binary_op_between_snowpark_columns` is getting worse IMO. Just a heads up, I will make some refactoring later after the recent changes to make it more structured.